test/e2e/sourcery.js

```diff
@@ -1,49 +1,118 @@
-var conf = require('../../nightwatch.conf.js');
+const c = require('../../libs/constants');
+const common = require('../../obj/common');
+const login = require('../../obj/login');
+const timeLogging = require('../../obj/timeLogging');
+const tasks = require('../../obj/tasks');
 
 module.exports = {
     'Login to sourcebooks': function (browser) {
+        let userUnderTest = "Lukas Klimišinas";
+        let userRole = "Admin";
+
+        browser
+            .url(browser.launchUrl)
+            .waitForElementVisible(common.pageTitle)
+            .isVisible(login.userSelect, function (result) {
+                if (result.status === c.ELEMENT_FOUND) {
+                    browser.click(login.userSelect)
+                }
+            })
+            .isVisible(login.getSpecificSelectUserOption(userUnderTest), function (result) {
+                if (result.status === c.ELEMENT_FOUND) {
+                    browser.click(login.getSpecificSelectUserOption(userUnderTest))
+                }
+            })
+            .assert.containsText(login.userSelectedValue, userUnderTest)
+            .isVisible(login.userRoleSelect, function (result) {
+                if (result.status === c.ELEMENT_FOUND) {
+                    browser.click(login.userRoleSelect)
+                }
+            })
+            .isVisible(login.getSpecificSelectUserRoleOption(userRole), function (result) {
+                if (result.status === c.ELEMENT_FOUND) {
+                    browser.click(login.getSpecificSelectUserRoleOption(userRole))
+                }
+            })
+            .assert.containsText(login.userRoleSelect, userRole)
+            .isVisible(common.submitButton, function (result) {
+                if (result.status === c.ELEMENT_FOUND) {
+                    browser
+                        .click(common.submitButton)
+                        .waitForElementVisible(timeLogging.loggedInUsersName);
+                }
+            })
+            .assert.containsText(timeLogging.loggedInUsersName, userUnderTest)
+            .waitForElementVisible(timeLogging.navbar)
+            .assert.containsText(timeLogging.navbar, timeLogging.navbarLinks)
+            .assert.containsText(timeLogging.navbarSelected, timeLogging.navbarSelectedValue)
+            .assert.cssProperty(timeLogging.navbarSelected, timeLogging.activeProperty, timeLogging.navbarActiveValue);
+    },
+
+    'Admin creates new task': function (browser) {
+        let ran = Math.random() * 100;
+        let ran1 = Math.random();
+        let taskName = ran * ran1;
+        let rate = 55;
+
+        browser
+            .url(browser.launchUrl + tasks.url)
+            .waitForElementVisible(common.pageTitle)
+            .isVisible(common.clickButton, function (result) {
+                if (result.status === c.ELEMENT_FOUND) {
+                    browser.click(common.clickButton)
+                }
+            })
+            .waitForElementVisible(common.pageTitle)
```

Comment:
You do not need to add so much .waitForElementVisible. You can check the first element/field/title after coming to new page and then you can assume that all form is displayed because of one of the items is already on the page.